server/src/test/java/org/cloudfoundry/identity/uaa/authentication/BackwardsCompatibleTokenEndpointAuthenticationFilterTest.java

```diff
@@ -167,8 +162,8 @@ public void attempt_jwt_token_authentication() throws Exception {
         verify(filter, times(1)).attemptTokenAuthentication(same(request), same(response));
         ArgumentCaptor<ExternalOAuthCodeToken> authenticateData = ArgumentCaptor.forClass(ExternalOAuthCodeToken.class);
         verify(externalOAuthAuthenticationManager, times(1)).authenticate(authenticateData.capture());
-        verifyZeroInteractions(passwordAuthManager);
-        verifyZeroInteractions(externalOAuthAuthenticationManager);
+        verifyNoInteractions(passwordAuthManager);
+        verifyNoMoreInteractions(externalOAuthAuthenticationManager);
```

Comment:
why not same replacement verifyNoInteractions vs. verifyNoMoreInteractions

Reply:
because we have line 164 `verify(externalOAuthAuthenticationManager, times(1)).authenticate(authenticateData.capture());` . The mock `externalOAuthAuthenticationManager` was used one time before. `verifyNoInteractions` - it's "strict mode". 

javadoc verifyNoMoreInteractions: `Checks if any of given mocks has any unverified interaction.
You can use this method after you verified your mocks - to make sure that nothing else was invoked on your mocks.`

Reply:
ok